vscode/grammars/ruby.cson.json

```
@@ -75,27 +76,24 @@
         "2": {
           "name": "entity.name.type.module.ruby"
         },
-        "3": {
-          "name": "entity.other.inherited-class.module.first.ruby"
-        },
-        "4": {
-          "name": "punctuation.separator.inheritance.ruby"
-        },
         "5": {
-          "name": "entity.other.inherited-class.module.second.ruby"
-        },
-        "6": {
-          "name": "punctuation.separator.inheritance.ruby"
-        },
-        "7": {
-          "name": "entity.other.inherited-class.module.third.ruby"
+          "name": "punctuation.separator.namespace.ruby"
+        }
+      },
+      "match": "(module)\\s*(([a-zA-Z0-9_]+)((::)[a-zA-Z0-9_]+)*)",
+      "name": "meta.module.ruby"
+    },
+    {
+      "captures": {
+        "1": {
+          "name": "keyword.control.class.ruby"
         },
-        "8": {
+        "2": {
           "name": "punctuation.separator.inheritance.ruby"
         }
       },
-      "match": "(?x)\n^\\s*(module)\\s+\n(\n  ([A-Z]\\w*(::))?\n  ([A-Z]\\w*(::))?\n  ([A-Z]\\w*(::))*\n  [A-Z]\\w*\n)",
-      "name": "meta.module.ruby"
+      "match": "(class)\\s*(<<)\\s",
```

```suggestion
      "match": "(class)\\s*(<<)\\s*",
```
Since `class<<self` is valid.